R/remove_funs.R

```diff
@@ -25,26 +25,19 @@ gsubAll <- function(string, old, new) {
 
 
 ### splits the string only on the first or the last occurrence of the separator
-### uses no longer package "stringr"
 halveString <- function (string, pattern, first = TRUE, colnames=c("X1", "X2") )  {
   if( !(inherits(colnames, "character") && length(colnames)==2 && length(unique(colnames)) == 2 )) {stop("'colnames' must be a unique character vector of length 2.")}
-  allSplit<- strsplit(x = string, split = pattern)
-  if(first)  {
-    ret <- as.matrix(data.frame(X1 = unlist(lapply(allSplit, FUN = function (l) { l[1]})),
-                                X2 = unlist(lapply(1:length(allSplit), FUN = function (l) {
-                                  if(length(allSplit[[l]]) == 1 )  { ret <- NA } else {ret <- substring(string[l], first = nchar(allSplit[[l]][1])+1+nchar(pattern) ) }
-                                  return(ret)})), stringsAsFactors = FALSE))
-  }  else  {
-    ret <- as.matrix(data.frame(X1 = unlist(lapply(1:length(allSplit), FUN = function (l) {
-                                     if (length(allSplit[[l]]) > 1 ) {
-                                         val <- substr(string[l],1,nchar(string[l])-(nchar(allSplit[[l]][length(allSplit[[l]])])+nchar(pattern)))
-                                     }  else  {
-                                         val <- string[l]
-                                     }
-                                     return(val)})),
-                                X2 = unlist(lapply(allSplit, FUN = function (l) {
-                                  if(length(l)==1) { ret <- NA}  else { ret <- l[length(l)]}
-                                  return(ret)})), stringsAsFactors = FALSE))
+  if(!first) {
```

Comment:
LGTM
Even though I do not understand the exact details, the code looks much leaner and easier to understand compared to before!

Reply:
I've tried to merge the changes into master, but it seems to fail ...

Reply:
I think the merging was successful, but you committed some later changes into branch `halveString` again? If you want, we could resolve this via zoom/phone.

Reply:
Yes, let's talk on the phone. I'm in the office ...